packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
```

Comment:
https://tc39.github.io/ecma262/#sec-ordinaryset

This combines steps `3.c.i` (overwriting `desc` with a new descriptor) and the steps under `4`. Looks fine.